v2/errors.go

```diff
@@ -0,0 +1,22 @@
+package dht
+
+import "fmt"
+
+// A ConfigurationError is returned when a component's configuration is found to be invalid or unusable.
```

Comment:
I think we should be using typed errors throughout so I brought this over from go-kademlia